### PR TITLE
chore: merge `next` into `dev`

### PR DESCRIPTION
**Pull request by lucasfernog**
### What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [ ] No

### Checklist
- [ ] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [ ] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [ ] I have added a convincing reason for adding this feature, if necessary

### CODE REVIEW

**Comment by lucasfernog** [comment]
This PR will be merged with a merge commit to preserve all `next` commits with authors.

File: .github/workflows/covector-version-or-publish-next.yml
Comment (amrbashir) [inline .github/workflows/covector-version-or-publish-next.yml:1]: shouldn't this file be removed now and instead port the changes to `covector-version-or-publish.yml`?

File: core/tauri/src/asset_protocol.rs
Comment (amrbashir) [inline core/tauri/src/asset_protocol.rs:5]:
```suggestion
#![cfg(feature = "protocol-asset")]
use crate::path::SafePathBuf;
```

File: tooling/cli/src/build.rs
Comment (amrbashir) [inline tooling/cli/src/build.rs:146]: Did we move this to the shell plugin?

**Review by amrbashir — CHANGES_REQUESTED** [review]

File: tooling/cli/src/build.rs
Comment (lucasfernog) [reply tooling/cli/src/build.rs:146]: not yet, we still need to figure out a safe way to handle this since the env var is set on the CLI.

File: .github/workflows/covector-version-or-publish-next.yml
Comment (lucasfernog) [reply .github/workflows/covector-version-or-publish-next.yml:1]: Yeah, I wasn't sure if I should do this right now or after merging the PR.

File: core/tauri/src/asset_protocol.rs
Comment (lucasfernog) [reply core/tauri/src/asset_protocol.rs:5]: Gotta be a PR instead since we can't push directly to next 😢

File: .github/workflows/covector-version-or-publish-next.yml
Comment (amrbashir) [reply .github/workflows/covector-version-or-publish-next.yml:1]: yeah lets do this modifications afterwards

File: tooling/cli/src/build.rs
Comment (amrbashir) [reply tooling/cli/src/build.rs:146]: how about adding it in the plugin build script?

**Review by amrbashir — APPROVED** [review]

File: tooling/cli/src/build.rs
Comment (lucasfernog) [reply tooling/cli/src/build.rs:146]: Yeah but to pass a security review it would have to be specific to that env var I think

File: core/tauri/src/asset_protocol.rs
Comment (lucasfernog) [reply core/tauri/src/asset_protocol.rs:5]: Actually probably don't even need this one since the feature flag is used on the module declaration

File: tooling/cli/src/build.rs
Comment (amrbashir) [reply tooling/cli/src/build.rs:146]: well, the build script could read the `TAURI_CONFIG` env var, validate the scope using this same logic and then set the env var if appropriate

Comment (lucasfernog) [reply tooling/cli/src/build.rs:146]: the build script is running on its own process though, so it actually needs a way to export the env var to tauri instead. something like what cargo does with the cargo:something printed to stdout.

File: core/tauri/src/asset_protocol.rs
Comment (amrbashir) [reply core/tauri/src/asset_protocol.rs:5]: yeah, this is just mainly to ensure that if the module declaration flag ever get deleted by accident, it will fail to build.

File: tooling/cli/src/build.rs
Comment (amrbashir) [reply tooling/cli/src/build.rs:146]: oh TIL.

Alright, I will keep this on my TODO list so I don't forget so we can discuss it later.

**Pull request merged**
